### ISSUE TITLE
profile (and thumbnail) marking for "streaks" when you have published for X consecutive days or weeks

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#319]
Github has this idea of "streaks" when you have published code for X consecutive days.

What if, along with the profile tagging (#104) and the idea of showing peoples' `role` next to their comment bio pics, we showed a "fire" icon if they've posted for X consecutive weeks? I.e. "you're on fire!!" This could similarly use [a Bootstrap tooltip](http://getbootstrap.com/javascript/#tooltips) to explain what it means, and potentially link to a page explaining it.

The code for the image on comments is currently at https://github.com/publiclab/plots2/blob/master/app/views/notes/_comment.html.erb#L10-L12

We could add a user.streak method which counts posts, comments, and wiki edits in the past week, and a class="streak" css to https://github.com/publiclab/plots2/blob/master/app/views/notes/_comment.html.erb#L9-L13

also, a hover Bootstrap tooltip and a link to explain what streaks are.

Actually there is a tally function which accepts a # of weeks already: https://github.com/publiclab/plots2/blob/master/app/models/user.rb#L134-L148

**Comment by ebarry** [comment]
+1

**Comment by btbonval** [comment]
how are we defining post/publish? Do comments count?

I think comments should count.

**Comment by jywarren** [comment]
could do; just needs to be easily queried so we can do it whenever we
"show" a user, like in comments.

**Comment by jywarren** [comment]
We could add a `user.streak` method which counts posts, comments, and wiki edits in the past week, and a `class="streak"` css to https://github.com/publiclab/plots2/blob/master/app/views/notes/_comment.html.erb#L9-L13

also, a hover Bootstrap tooltip and a link to explain what streaks are.

Actually there is a tally function which accepts a # of weeks already: https://github.com/publiclab/plots2/blob/master/app/views/notes/_comment.html.erb#L9-L13

**Comment by ananyo2012** [comment]
@jywarren Considering each post would have quite many number of comments finding streak for each user using a `user.streak` method would involve a considerable number of queries as we have to find the daily/weekly comment,wiki edits,note count.It would increase the page load time.For notes containing small number of comments it would not affect much but for those having large number of comments it would affect the performance of page load. So it would be better if we would show the streaks on individual profile page similar to that of github along with specific emphasis as the fire icon when the user a streak of X consecutive days. What are your views on it?

**Comment by ananyo2012** [comment]
@jywarren Ok I found a better way to do it. Instead of generating the daily comment/note count we could just find the no. of consecutive. That way we could improve the performance of page load. Should we also show the no. of comments,notes and wiki edits in that interval?

**Comment by jywarren** [comment]
Yes, i see how it could be a difficult query. Perhaps we could cache the
query to be run only once per day, or something, and flush the cache when
new content is posted?

In your better way (can you share a code snippet or pseudocode?) are you
saying you just collect all notes in the past X time, and see if there are
Y posts per day in that query?

**Comment by ananyo2012** [comment]
@jywarren Please see the explanation in #539

**Comment by ananyo2012** [comment]
@jywarren Well this Issue already is related to User streaks. But the ideas have changed over time. Making a new issue would lead to a redundancy? So should I make a new one on User streaks and then this should be closed or we can update this one?

**Comment by jywarren** [comment]
We could close this one and link to a new one which briefly summarizes the current state -- that way it'll be more readable but still linked here. 

But also, we might consider that since this requires some overall design thinking, not really just coding, that we could leave it as-is (or with a brief status summary at top) since it might be too early for a coder to take on.

**Comment by ananyo2012** [comment]
Ok I will ensure that there is enough references regarding this.

**Comment by ananyo2012** [comment]
@jywarren I have created a new issue linking this at #699. We can perhaps close this now if you approve.

**Comment by jywarren** [comment]
Perfect!
